packages/design-system-react/src/components/button-base/ButtonBase.stories.tsx

```diff
@@ -118,6 +126,15 @@ export const Size: Story = {
   ),
 };
 
+export const TextProps: Story = {
+  args: {
+    children: 'Button with custom text variant',
+    textProps: {
+      variant: TextVariant.BodySm,
+    },
+  },
+};
+
```

Comment:
Adding TextProps story and overriding the text variant

<img width="1500" alt="Screenshot 2025-01-07 at 10 41 08 AM" src="https://github.com/user-attachments/assets/38264f13-1586-4e40-89e3-f10c79f14c45" />